### ISSUE TITLE
Horatio Changes

### ISSUE THREAD

**Issue by captaincobbs** [wakystuf/ESG-Mod#575]

**Comment by Arpharnator** [comment]
First thing I'll probably do is change the buildings of the 1st chapter that give bonuses based on major pops (ie the thing that gives +4 ind per major pop on a system) to be empire wide OR to put them in the empire unique tech in T4 empire dev and change the rewards

**Comment by Arpharnator** [comment]
Splices cost 
3 base and +3 for each extra, splicing rewards buffed accross the board, you no longer need to splice an entire galaxy to get good results, even tho each splicing will be harder you will benefit more, thus making you able to exploit the pop types of a localized place rather than going with 10 overcol, especially with the approval changes slamming the horatio massive dong. This also make conquest of populated areas a bit better due to being able to actually kill a lot of little shits for greater benefits, so that should help.

**Comment by Arpharnator** [comment]
An issue that still annoys me is how much col cap I feel you need to function as horatio because your colonies will be done somewhat fast but you don't get that much benefits on staying with a fairly tight number of systems (remember approval changes?) but you mostly have to, especially when you consider that you need quite a bit of place to be able to grow minors in your systems and then replace the losses.
Another problem that arises is simply the travel time of those pops, which can take a few turns in the early game and stays fairly long when you have a larger empire and you don't want to drop them off to the places that are supposed to be full of horatios.

**Comment by Arpharnator** [comment]
Horatio heroes :
T1:
-20 food and +3 per level -> 40  and +6
-+2 max probes, +100% regen rate, 1 move -> +4 max probes, 200% regen rate, 2 move
T2:
0.05 food to IDSI per sd level
defense skill? or maybe something about pods(doubt but still)? or something about perfection (double edged sword)
T3:
actual governing skill?
+3/6 extra luxury -- resource extractor kind of horatio
+25% module multiplier on modules/ships
T4:
50% ind cost increase, 25% defensive stats and damage increase
something that increases stats based on ship cost (make sure to introduce an initial ship cost property)

**Comment by Arpharnator** [comment]
thoughts?

**Comment by wakystuf** [comment]
@Arpharnator seems cool. My thoughts:
* What was the prior cost curve of splices?
* 40 food flat is a LOT for the early game. Maybe leave the fixed part alone and only increase the scaling part?
* I wouldn't mess with probe regen, but buffing the other effects seems fine. I'm having nightmares about people stacking that with the new probes law. I assume as Horatio you can simply prioritize the Signal and Lifeform curios that generate new minor pops, right?
* For T2 I think you want 1 of each admiral and governor, right? I think that's the pattern for every other faction (maybe there's an exception I'm forgetting about?)
* For T3, the vanilla lux boost is already really good, I think; I've intentionally recruited Horatio heroes before to put them in charge of my trade HQ's and things go boom (and that's even more true with the econ rework with buffed lux yields).
* The other T3 skill and the T4 skill both seem cool. I'm not sure if the T4 skill will be moddable, but you can try. The challenge will be attaching some permanent tag to each ship saying "Made by Horatio Governor", which may or may not be possible. Cool concept though if you can make it work

**Comment by Provolonage** [comment]
I like the idea of an alternate scaling for food since getting food splices or food buildings/ food minor traits feels bad, the number might be way too high since bhagba minor trait on ratio basically gives infinite influence, maybe switch the t2 food conversion with the luxury bonus so you can use the luxury bonus to improve expansion more early and gate the bonus idsi a bit, a defense skill that scaled off of splices somehow would be cool since you lose max manpower from losing pop so it could be a great way to counter that and lean into the horatio power fantasy a bit, but that seems like a pain to get to work programming wise, I'm kinda against giving ratio even more powerful but expensive ships, at some point the fleets will just auto win and get fed and that isnt fun to fight, how does the idea of making the t4 talent -5% cost on ship +5% damage on module +5% ship health sound, more of an improvement to the price of perfection trait maybe call it Actualized Excellence or something

**Comment by Arpharnator** [comment]
@wakystuf 
1) 2+2
2) you know of the overseer skill no? gives 40 at level 2 if you have a fertile and 60 if you have 2, not sure what the problem is
3) the probe regen only allows you to probe stuff without actually having a probe module on otherwise it takes forever, and that's just if you happen to need an extra curio revealer, regen is pretty standard. it's an investment that won't go into system skills
4) yea I know, I just don't know what to put for the admiral skill
5) yea but I removed the skill that made shit explode, this one is the flat one, it's a good skill but it has no governing value, requires your hero to be placed on shit systems sometimes and since the luxury explosion skill has been removed, it's not as good. Like I put, usually choosing this skill means that you gonna put the horatio on extraction duty and not generating actual fids, and for a T3 +2+4 was rather bad.
6) I'm also considering giving horatio a third T3 skill, to be able to get a proper governing skill on your good system instead of being awkardly stuck with a T3 that doesn't actually give fidsi. In fact it's not impossible the T2 gov skill becomes T3 and I add another T2 skill, since that T2 gov skill seems a bit good for T2.
I'll have to see how it goes for the ind spent on construction thing, I want to lean into the theme of expensive but good ships.
I'm also confident it can work since I already know I can borrow the effects of that T4 military building that gives 2 level and 10% ship hp.

@Provolonage 
I'll have to see about the baghaba splice, minors became much more expensive to get, laws are quite costly.
The luxury bonus is in T3 because it needs to be good to allow for luxury extraction of T3s in a decent amount.
My goal with the more expensive ships is that you want to use good system devs to be able to afford them using the T3 extraction skill.
It reminds me of card games where you play discard decks where you sacrifice your own hand to be able to put forth powerful cards, but turns out the card you just discarded had a powerful effect if you discarded it but its play effect is fairly shit.
In T4 the discarding effect being the bonus ship cost but better stats and the discarded card being the T4 admiral skill that increases by how much ind you put into the thing.

**Comment by Arpharnator** [comment]
update to the thing : 
T1:
-15 food and +2 per level -> 30 and +4
-+1 max probes, +100% regen rate, 1 move -> +2 max probes, 100% regen rate, 2 move
T2:
something here about governing
defense skill? or maybe something about pods(doubt but still)? or something about perfection (double edged sword)
T3:
0.05 food to IDSI per sd level
+3/6 extra luxury -- resource extractor kind of horatio
+25% module multiplier on modules/ships -- might change due to complications and the fact that it's very much damage oriented
T4:
50% ind cost increase, 25% defensive stats and damage increase
something that increases stats based on ship cost (make sure to introduce an initial ship cost property)

**Comment by wakystuf** [comment]
* Yeah the T1 is literally just a better version of the Vody skill (though realistically that Vody skill should probably get replaced with something more thematically appropriate). I guess that's fine then?
* if +100% regen only applies to the hero ship, then that's fine. I just wouldn't want it to go boom with hyperium probed-scouts + sci law
* Maybe just move the resource extraction skill to T2 but leave it at 2/4? 4 is really a lot of lux deposit value, especially since I'm nearly certain it also runs through the trade yields. So +4 on a good lux system with a trade HQ = boom; I've specifically recruited a Horatio hero before as Vodyani and it was delicious.
* The "better ships" thing seems...maybe fine? Just be careful since remember heroes can also be used against you. If the Cravers get their slimy hands on one of these guys, then just make sure that you aren't boned.

**Comment by Provolonage** [comment]
I feel like the probe thing is fine since the only time you'd ever get to use that early when it would matter is on your overseer and even then you might not want to so you can get more system skills, by the time you could get the horatio seeker most curiosity's will be taken and you wont need more then 2-4 probes a system so its just movement, honestly just making the move skill have 2 levels is nice so when you get two counselors from your quest you can use one as an admiral and not have to waste a point on a system skill to get level 3 skills, since you probably have gotten better system heroes from market and the academy, I feel like the original food skill was fine because clone wares gave approval and scaled better, the only time you'd really get good value out of it now is if you sat your overseer on your home system once it got level 3 and colonized a lot, otherwise its mostly overkill anyway and the lack of approval hurts. Food to 0.05 IDSI per system level seems too high, I feel like the skill that makes ships cost more but stronger should be a governing skill, I feel that making it system based would be unintuitive for a lot of players.

**Comment by Provolonage** [comment]
Also I like your reasoning for the higher ship cost so I think you should go ahead with it and we'll see if it needs tweaking

**Comment by Arpharnator** [comment]
@wakystuf 
1) Vodo shouldn't have food if they can't use it properly. Really strange that they get food when they get -50%
2) tbf the limit on horatio's scouting is more about the number of curios rather than the regen because when you get blue probe you regen instant anyway. Also there's no point in getting science law as horatio, and as this is an early game skill, if other factions manage to get a horatio hero that's not what they would use it for since it's kinda late.
3) I can see what you mean, but this is about heroes for horatio not for other factions, the same way getting a horatio can make you gain muscle, anybody getting a vodo hero gains muscle. You saying that the skill should be T2 and be 2/4 is kinda int if you consider that this helps just about every other faction in the game and kinda gimps horatio. The T3 comes later and requires actual investment from vodo instead of getting instant T2.
4) same point as above, and I also suspect that not so many factions could actually run the hero properly since it's quite the massive ind cost boost and not many factions can afford that much.

I see your worry about horatio heroes being too good for other factions, but when you consider that most of the factions can't really synergize well with the skills, it's less of an issue. For example the T3 food skill, at best only Unfallen can use properly, but in the meantime still have problems utilizing the other skills. Vodo might be able to use luxury extraction but they can't use any of the food skills properly. I have yet to design the T2 gov skill, and I have my idea for the T2 admiral skill.
Also reminder that the thing that probably made you go boom luxuries is the capstone, which has been removed here.

@Provolonage 
Clone wares had the problem of being useless as a T1 skill, giving % food bonus is usually quite worthless in the early game, especially as horatio where you either need a lot of food (new system) and % won't do much and give 4 food, or you already are swimming in food and don't have food to ind.
The extra ship cost and bonuses is actually a governing skill, it's like real time relays in its workings

**Comment by Arpharnator** [comment]
Horatio heroes :
T1:
-15 food and +2 per level -> 30  and +4
-+1 max probes, +100% regen rate, 1 move -> +2 max probes, 100% regen rate, 2 move
T2:
20% inf, 3 inf per pop
+50% ship manpower dmg absorption, +30% crew dmg bonus
T3:
0.05 food to IDSI per sd level
+3/6 extra luxury -- resource extractor kind of horatio
+20% hull palting +40% shield absorption
T4:
50% ind cost increase, 25% defensive stats and damage increase
gain 6 shield and 3 health per ind spent at ship creation

**Comment by wakystuf** [comment]
Yeah, seems fine mostly. I probably wouldn't buff the lux extraction, but I'm not going to fall on my sword over it (though I would then probably pick a different reward other than -SD cost % for the quest, since cheaper SD + lux buff = boom)

**Comment by Arpharnator** [comment]
my reasoning for the quest stuff was that you have a setup part and then the actual reward, ie you have sd cost reduction to setup sd levels and then the actual reward tied to sd.
So if you don't wanna have that reward then you need to find me a new one that's tied with SD or if you want you can also find me a new reward for both :D :D

**Comment by wakystuf** [comment]
Instead of SD cost I would just give them...literally anything else? A few examples:
* Passive food to IDS
* Lower cost of converting non-bald pops into bald (and/or making it not block the queue)
* Faster civilian ships + higher migration score
* Slightly lower splicing cost

**Comment by Arpharnator** [comment]
it's all cringeeeeeeee

**Comment by Arpharnator** [comment]
and shit

**Comment by wakystuf** [comment]
I mean you can't say they are all shit in the abstract; clearly there is some % of Food to IDS that is not shit, and similar for the splicing cost I would think?. I mean, if 5% food to IDS is shit, then what about 10%? 100%? 1000000%? Surely there is a number where it's appropriate (and similar for the splicing).

**Comment by Arpharnator** [comment]
the first quest gives the buildings. They are made empire wide but worse
All 1.2 quests give Five -> done


Chapter 3.1 Religious gets +75% hero xp, horatio hero level 5
Chapter 3.1 Ecologist gets +1 pop slots on fertile, horatio hero level 5
Chapter 3.1 Industrialist gets -30% on sd costs, horatio hero level 5

Chapter Eco/Ind/Religious 3.2 gets a Horatio hero level 8 alongside their respective building -> done

Chapter Eco 4 gets +15% FIDSI on full planets 	
Chapter Religious 4 gets Systems with heroes get +1.5% FIDSI per hero level and +2% border growth per hero level
		         Hero ships gain 5% dmg and health per level alongside an extra +1 module multiplicator
Chapter Ind 4 gets +100% consumption, +7% FIDSI per sd level

**Comment by captaincobbs** [comment]
Bro these rewards are bonkers

**Comment by Arpharnator** [comment]
it's a big stick for a suffering faction till the colcap changes

**Comment by captaincobbs** [comment]
why not just wait until colcap changes

**Comment by Arpharnator** [comment]
no cuz I still needed to change quest

**Comment by captaincobbs** [comment]
Not accepting, the rewards are way too much

**Comment by Arpharnator** [comment]
if you want to talk about it go play horatio

**Comment by Arpharnator** [comment]
otherwise I'm not takng feedback from people not playing ratio

**Comment by Arpharnator** [comment]
I did all this work in knowledge of what my faction does, so I'm going to take someone's word who doesn't play the faction that it needs to change

**Comment by Arpharnator** [comment]
changes :
exp from religious reward : 40% from 75%
dev sys cost : 20% to 30%
religious fidsi from hero reward : 0.75% FIDSI per hero level from 1.5%
ind system dev fidsi : 4% and 50% consumption from 7% and 100% consumption

**Comment by wakystuf** [comment]
@Arpharnator I need to load up the game to look at the quest part some more, but this mostly makes sense. I made a few small tweaks to your code; let me know if any of those seem bad to you and we can talk about them. In the meantime, a few questions:
1) Does the MP fleet skill work properly? It's tricky to test, so just wanted to make sure.
2) Does the ind to health and shield work? That's a cool mechanic, but fairly new. With my stuff I always find that the cooler/newer it is then the more likely it is to break, so just wanted to make sure.
3) Does the Food to IDSI conversioni cause any circularity issues with other stuff into food (ex: minor faction trait)? I remember when I tried to do something like that with the UE it took me like 5 fucking tries to make it not crash the game.
4) I saw you added the ShipConstructionEffectFromHeroSkillShipHoratioSkill infrastructure. Is that used for anything, or was that a prior idea that got removed?
5) Am I reading correctly that one of the quest "rewards" provides an *increase* in food consumption? Is that intended?
6) The only hero skill that I thought was a problem was the 50% reduction in ship cost--that's just so so much. I get that it's a capstone, but...damn. Maybe we try to find another solution on that issue.

**Comment by Arpharnator** [comment]
the tweaks I agree
1) the mp damage boost works
2) the ind to health works
3) don't believe so, I have yet to get that problem
4) it is used for the final horatio skill governor to give the bonus stats
5) yes it is intended, u know how to read
6) it's an increase in ship cost to get the bonuses in 4)

**Comment by wakystuf** [comment]
OK, sounds good. Can you test #3 to make sure? Trigger the skill and assimilate a minor that gives ind to food or something and see if the game explodes. That's what happened to me with the UE stuff. The game can be very rude at times.

**Comment by wakystuf** [comment]
OK, I just fixed something in ClassShip when I was trying to understand your answer on point 4. In any SimulationDescriptor, the game gets grumpy for some reason if you put a modifier with the properties, so I moved the modifer down to the modifiers section.

Also, are you suuuuure that the ShipConstructionEffectFromHeroSkillShipHoratioSkill is getting used for that final hero skill? To me it looks like that code is an Admiral skill that applies the extra health and shield if the hero is on the fleet. In order to work as a governor ability it would need to apply a tag to the ships produced in its system (specifcally the tag ShipConstructionEffectFromHeroSkillShipHoratioSkill).

**Comment by Arpharnator** [comment]
100% sure

**Comment by Arpharnator** [comment]
I remember testing it

**Comment by Arpharnator** [comment]
@wakystuf 
the conversion stuff works properly even with another conversion

**Comment by Arpharnator** [comment]
proof of last skill working
before skill
![image](https://user-images.githubusercontent.com/49691625/131585434-572e3c07-2cbb-4936-9ab9-e611b19723cc.png)
after skill
![image](https://user-images.githubusercontent.com/49691625/131585455-aa852dea-ebab-4595-8e1d-3a906b573dfd.png)

**Comment by wakystuf** [comment]
Oh, I see now. I was referring to the other admiral skill. I get it now. Cool. 

I'll probably fiddle with some tooltips later tonight but mostly seems good

**Comment by wakystuf** [comment]
![image](https://user-images.githubusercontent.com/79942510/131596969-0dda4784-2707-4d65-8103-f1fe04d7a311.png)
@Arpharnator This is what I was worried about. This plus the Amoeba faction trait = game dies (it literally hard crashed for me)

**Comment by wakystuf** [comment]
![image](https://user-images.githubusercontent.com/79942510/131602867-d23f26a9-fd9e-46f4-a885-a96f93f2bdcc.png)
@Arpharnator I think this tooltip is the only other one we need to fix. Once that's fixed plus the circularity issue is resolved somehow (changing the way the math works, priorities, or however you want to do it) then I think this should be all set? I can't think of anything else that still needs to be done before we merge, unless someone yells about the summary that I posted to the senate

**Comment by captaincobbs** [comment]
What's the exact effect?

**Comment by wakystuf** [comment]
```xml
<SimulationDescriptor Name="EmpireImprovementQuestHoratioESG_5" Type="EmpireImprovement">
    <BinaryModifier TargetProperty="SystemGrowth" Operation="Percent" Left="0.0075" BinaryOperation="Multiplication" Right="$(AssignedHeroLevel)" Path="ClassEmpire/ClassColonizedStarSystem" SearchValueFromPath="true"/>
    <BinaryModifier TargetProperty="SystemProduction" Operation="Percent" Left="0.0075" BinaryOperation="Multiplication" Right="$(AssignedHeroLevel)" Path="ClassEmpire/ClassColonizedStarSystem" SearchValueFromPath="true"/>
    <BinaryModifier TargetProperty="SystemMoney" Operation="Percent" Left="0.0075" BinaryOperation="Multiplication" Right="$(AssignedHeroLevel)" Path="ClassEmpire/ClassColonizedStarSystem" SearchValueFromPath="true"/>
    <BinaryModifier TargetProperty="SystemResearch" Operation="Percent" Left="0.0075" BinaryOperation="Multiplication" Right="$(AssignedHeroLevel)" Path="ClassEmpire/ClassColonizedStarSystem" SearchValueFromPath="true"/>
    <BinaryModifier TargetProperty="SystemEmpirePoint" Operation="Percent" Left="0.0075" BinaryOperation="Multiplication" Right="$(AssignedHeroLevel)" Path="ClassEmpire/ClassColonizedStarSystem" SearchValueFromPath="true"/>
    <Modifier TargetProperty="Multiplier" Operation="Addition" Value="1" Path="ClassEmpire/ClassGarrisonFleet/ClassShip,ShipRoleHero/ClassSection/ClassModule"/>
    <Modifier TargetProperty="TriggerEmpireImprovementReligiousHoratio" Operation="Force" Value="1" Path="ClassEmpire//ClassHero" TooltipHidden="true"/>
  </SimulationDescriptor>
```

**Comment by captaincobbs** [comment]
I would hide the tooltip and make one of them have an override of +7.5% FIDSI per Governor Level on Systems

**Comment by Arpharnator** [comment]
problem then is that you don't get to see how much it gives you on the system
It's better to be like that than not seeing how much you're getting on the system screen, so I wouldn't change it

**Comment by Arpharnator** [comment]
changed the religious ratio reward to be 1% to not have to worry over tooltips and because it won't make or break the reward, in  fact it prolly was a little weak with the removal of the inf radius

**Comment by Arpharnator** [comment]
I believe that should be about it

**Comment by wakystuf** [comment]
Yeah that's a good point on the tooltip. I kept it mostly as is and just updated the module multiplier one. I need to test some stuff for the Federation changes plus add the effects that Cobbs forgot about (SD cost and approval from out of power parties) and then I'll merge later today probably
